GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -955,12 +955,10 @@ private void InternalInitialize()
 
                     if (AppSettings.ShowAheadBehindData)
                     {
-                        bool isAheadBehindDisplayedForAllBranches = !MainSplitContainer.Panel1Collapsed && (AppSettings.RepoObjectsTreeShowRemotes || AppSettings.RepoObjectsTreeShowBranches);
                         string currentBranch = RevisionGrid.CurrentBranch.Value;
                         ThreadHelper.FileAndForget(async () =>
                         {
-                            string aheadBehindBranch = isAheadBehindDisplayedForAllBranches ? "" : currentBranch;
-                            IDictionary<string, AheadBehindData> aheadBehindData = _aheadBehindDataProvider?.GetData(aheadBehindBranch);
+                            IDictionary<string, AheadBehindData> aheadBehindData = _aheadBehindDataProvider?.GetData(currentBranch);
```

Comment:
This is not something I prefer, it is OK to wait a little for this.
But I will not try block if the reason is well described in a comment.

Parallel git-for-each-ref will slow each other down.
Repeating that this info should not be requested until revisions are being shown (spinner is removed, Loading is shown in upper right corner of the grid.

Reply:
> This is not something I prefer, it is OK to wait a little for this.

As explained, this case will be beneficial when you have a lot of branches (maybe more than 2k) so in this exact case, you won't wait a little but maybe more than 1 or 2s (in my test repo it's more than 6s). So I thinks that better to not wait.

If you have less than this number of branches, in fact, whatever you pass as arguments (empty or current branch), the behavior will be the same and value will be taken from cache because the data retrieve is triggered first from left panel.

> Parallel git-for-each-ref will slow each other down.

And when you have a lot of branches and so there will 2 commands run, the first short lived one is already finished when the 2nd is launched.
So really no problem here....

As a summary, in normal case (not too much branch), the parameter is useless. And when we have a huge number of branches, it allows to get current branch info without slowing down retrieving data for all branches that happens later.

Reply:
@gerhardol have a look at the code comment to see if it is good enough...

Reply:
I can live with this.
But I will be annoyed next time I look at git commands and try to find what is not really needed and use a minute of my time to find the comment.

Reply:
I pushed a simple test case. @pmiossec merge if OK, you updated the commit message anyway

Reply:
> But I will be annoyed next time I look at git commands and try to find what is not really needed and use a minute of my time to find the comment.

It's what I try to explained, maybe not clearly enough, that in a normal size repository, the command **is not run at all** because the values are taken from the  filled dictionary because the retrieve from left panel that fetch **all** the data is done first. And the command is only run in huge repositories where it improve the UX a lot and without the 2 commands being run at the same time.

So you won't be annoyed 😉

Reply:
I will look at the log and try to optimize, will stop at the comment. So still some annoyed. But I would be more annoyed by running this alone.
Thanks for cooperation!